llvm/lib/Analysis/Loads.cpp

```diff
@@ -171,7 +169,7 @@ static bool isDereferenceableAndAlignedPointer(
                                               Size, DL, CtxI, AC, DT, TLI,
                                               Visited, MaxDepth);
 
-  if (CtxI && (!UseDerefAtPointSemantics || !V->canBeFreed())) {
+  if (CtxI && !V->canBeFreed()) {
```

Comment:
Might be worth a comment here emphasizing the restriction on the dereferenceable result from the assume.

Reply:
Added + a TODO to extend to check between assume & use, thanks